### PR TITLE
Windows Path on Bootstrap

### PR DESCRIPTION
**Pull request by Xorima**
Fix issue with windows path during bootstrap

## Description
As a windows user when I bootstrap a node I expect the path to not have items removed as my cookbooks may depend on executables not stored on common system paths, such as sqlcmd which will break and cause the initial chef run to fail

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tas50** [comment]
Another one for you @marcparadise

**Comment by Xorima** [comment]
I think the reason this was changed was too long file paths but it has other unintended causes, 
I guess the question is how can we validate the length of the path without it getting too long, and as we are using a cmd file it limits the options we would have with say powershell

File: lib/chef/knife/core/windows_bootstrap_context.rb [hunk 1]
```diff
@@ -158,7 +158,7 @@ def get_log_location
 
         def start_chef
           bootstrap_environment_option = bootstrap_environment.nil? ? "" : " -E #{bootstrap_environment}"
-          start_chef = "SET \"PATH=%SystemRoot%\\system32;%SystemRoot%;%SystemRoot%\\System32\\Wbem;%SYSTEMROOT%\\System32\\WindowsPowerShell\\v1.0\\;C:\\ruby\\bin;C:\\opscode\\chef\\bin;C:\\opscode\\chef\\embedded\\bin\"\n"
+          start_chef = "SET \"PATH=%PATH%;C:\\ruby\\bin;C:\\opscode\\chef\\bin;C:\\opscode\\chef\\embedded\\bin\"\n"
```
Comment (marcparadise) [inline lib/chef/knife/core/windows_bootstrap_context.rb]: By allowing custom path elements first, we run the risk of grabbing the wrong ruby/other utility.  Could you modify this to something like: 
`"SET \"PATH=%SystemRoot%\\system32;%SystemRoot%;%SystemRoot%\\System32\\Wbem;%SYSTEMROOT%\\System32\\WindowsPowerShell\\v1.0\\;C:\\ruby\\bin;C:\\opscode\\chef\\bin;C:\\opscode\\chef\\embedded\\bin\;%PATH%"\n"
```
It'll have some redundancy, but will ensure that the chef-related paths are first during bootstrap.

**Comment by Xorima** [comment]
Resolved as per feedback @marcparadise apologies for the delay

**Comment by Xorima** [comment]
Resolved as per feedback @marcparadise apologies for the delay

**Comment by Xorima** [comment]
tests failing on linux so doubt it is due to this one?

**Comment by Xorima** [comment]
Nudge @marcparadise could this be looked at please?

**Pull request merged**